File: proxyipcenter-service/src/main/java/com/virjar/entity/Proxy.java
package com.virjar.entity;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Proxy {
    private Long id;

    private String ip;

    private String proxyIp;

    private Integer port;

    private Long ipValue;

    private String country;

    private String area;

    private String region;

    private String city;

    private String isp;

    private Integer countryId;

    private Integer areaId;

    private Integer regionId;

    private Integer cityId;

    private Integer ispId;

    private Long addressId;

    private Byte transperent;

    private Long speed;

    private Byte type;

    private Long connectionScore;

    private Long availbelScore;

    private Date connectionScoreDate;

    private Date availbelScoreDate;

    private Date createtime;

    private Boolean supportGfw;

    private Long gfwSpeed;

    private String source;

    private String crawlerKey;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getIp() {
        return ip;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public String getProxyIp() {
        return proxyIp;
    }

    public void setProxyIp(String proxyIp) {
        this.proxyIp = proxyIp;
    }

    public Integer getPort() {
        return port;
    }

    public void setPort(Integer port) {
        this.port = port;
    }

    public Long getIpValue() {
        return ipValue;
    }

    public void setIpValue(Long ipValue) {
        this.ipValue = ipValue;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getArea() {
        return area;
    }

    public void setArea(String area) {
        this.area = area;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getIsp() {
        return isp;
    }

    public void setIsp(String isp) {
        this.isp = isp;
    }

    public Integer getCountryId() {
        return countryId;
    }

    public void setCountryId(Integer countryId) {
        this.countryId = countryId;
    }

    public Integer getAreaId() {
        return areaId;
    }

    public void setAreaId(Integer areaId) {
        this.areaId = areaId;
    }

    public Integer getRegionId() {
        return regionId;
    }

    public void setRegionId(Integer regionId) {
        this.regionId = regionId;
    }

    public Integer getCityId() {
        return cityId;
    }

    public void setCityId(Integer cityId) {
        this.cityId = cityId;
    }

    public Integer getIspId() {
        return ispId;
    }

    public void setIspId(Integer ispId) {
        this.ispId = ispId;
    }

    public Long getAddressId() {
        return addressId;
    }

    public void setAddressId(Long addressId) {
        this.addressId = addressId;
    }

    public Byte getTransperent() {
        return transperent;
    }

    public void setTransperent(Byte transperent) {
        this.transperent = transperent;
    }

    public Long getSpeed() {
        return speed;
    }

    public void setSpeed(Long speed) {
        this.speed = speed;
    }

    public Byte getType() {
        return type;
    }

    public void setType(Byte type) {
        this.type = type;
    }

    public Long getConnectionScore() {
        return connectionScore;
    }

    public void setConnectionScore(Long connectionScore) {
        this.connectionScore = connectionScore;
    }

    public Long getAvailbelScore() {
        return availbelScore;
    }

    public void setAvailbelScore(Long availbelScore) {
        this.availbelScore = availbelScore;
    }

    public Date getConnectionScoreDate() {
        return connectionScoreDate;
    }

    public void setConnectionScoreDate(Date connectionScoreDate) {
        this.connectionScoreDate = connectionScoreDate;
    }

    public Date getAvailbelScoreDate() {
        return availbelScoreDate;
    }

    public void setAvailbelScoreDate(Date availbelScoreDate) {
        this.availbelScoreDate = availbelScoreDate;
    }

    public Date getCreatetime() {
        return createtime;
    }

    public void setCreatetime(Date createtime) {
        this.createtime = createtime;
    }

    public Boolean getSupportGfw() {
        return supportGfw;
    }

    public void setSupportGfw(Boolean supportGfw) {
        this.supportGfw = supportGfw;
    }

    public Long getGfwSpeed() {
        return gfwSpeed;
    }

    public void setGfwSpeed(Long gfwSpeed) {
        this.gfwSpeed = gfwSpeed;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getCrawlerKey() {
        return crawlerKey;
    }

    public void setCrawlerKey(String crawlerKey) {
        this.crawlerKey = crawlerKey;
    }
}

File: proxyipcenter-dotnet/proxyipcenter-dotnet/Model/Proxy.cs
//============================================================
//http://codelover.link author:李国宝
//============================================================
//============================================================
//此处的属性名请与json包中对应的名字保持一致 否则解析会非常困难
//============================================================
using System;
using System.Collections.Generic;
using System.Text;

namespace proxyipcenter_dotnet.Model
{	

	[Serializable()]
    
    /// <summary>
    /// 
    /// </summary>
	public class Proxy
    {

       
        #region 属性

        /// <summary>
        /// 主键
        /// </summary>
		public long id{get;set;}
            
        /// <summary>
        /// IP地址
        /// </summary>
		public string ip{get;set;}
            
        /// <summary>
        /// 代理IP，也就是目标网站最终看到的IP（多级代理的情况ip和proxy_ip不会相同）
        /// </summary>
		public string proxyIp { get;set;}
            
        /// <summary>
        /// 端口号
        /// </summary>
		public int? port{get;set;}
            
        /// <summary>
        /// ip的数字表示，用于过滤连续IP问题
        /// </summary>
		public long? ipValue { get;set;}
            
        /// <summary>
        /// 国家
        /// </summary>
		public string country{get;set;}
            
        /// <summary>
        /// 地区
        /// </summary>
		public string area{get;set;}
            
        /// <summary>
        /// 省
        /// </summary>
		public string region{get;set;}
            
        /// <summary>
        /// 市
        /// </summary>
		public string city{get;set;}
            
        /// <summary>
        /// 运营商
        /// </summary>
		public string isp{get;set;}
            
        /// <summary>
        /// 国家代码
        /// </summary>
		public string countryId{ get;set;}
            
        /// <summary>
        /// 地区代码
        /// </summary>
		public string areaId{ get;set;}
            
        /// <summary>
        /// 省级代码
        /// </summary>
		public string regionId{ get;set;}
            
        /// <summary>
        /// 城市代码
        /// </summary>
		public string cityId{ get;set;}
            
        /// <summary>
        /// isp代码
        /// </summary>
		public string ispId{ get;set;}
            
        /// <summary>
        /// 地理位置ID，融合各个地理位置获取的一个数字，数值约接近表示实际地理位置约接近
        /// </summary>
		public long? addressId{get;set;}
            
        /// <summary>
        /// 透明度(高匿，普通，透明)
        /// </summary>
		public sbyte? transperent{get;set;}
            
        /// <summary>
        /// 连接时间（越小速度越快）
        /// </summary>
		public long? speed{get;set;}
            
        /// <summary>
        /// 类型（http，https,httpAndHttps,socket,qq）
        /// </summary>
		public sbyte? type{get;set;}
            
        /// <summary>
        /// 连接性打分
        /// </summary>
		public long connectionScore { get;set;}
            
        /// <summary>
        /// 可用性打分
        /// </summary>
		public long availbelScore { get;set;}
            
        /// <summary>
        /// 连接性打分时间
        /// </summary>
		public long connectionScoreDate { get;set;}
            
        /// <summary>
        /// 可用性打分时间
        /// </summary>
		public long availbelScoreDate { get;set;}
            
        /// <summary>
        /// 收录时间
        /// </summary>
		public long createtime { get;set;}
            
        /// <summary>
        /// 是否支持翻墙
        /// </summary>
		public sbyte? support_gfw{get;set;}
            
        /// <summary>
        /// 翻墙访问速度
        /// </summary>
		public long? gfw_speed{get;set;}
            
        /// <summary>
        /// 资源来源url
        /// </summary>
		public string source{get;set;}
            
        /// <summary>
        /// 爬虫key，用户统计爬虫收集分布
        /// </summary>
		public string crawler_key{get;set; }


        #endregion


        #region

        
        public Proxy()
        {

        }

        #endregion
    }
}
